File: server/src/domain/dtos/users/update-user.dto.ts
export class UpdateUserDto {
    private constructor(
        public readonly name: string,
        public pictureUrl: string,
        public pictureId: string,
    ) { }

    static create(props: { [key: string]: string }): [string?, UpdateUserDto?] {
        const { name, pictureUrl, pictureId } = props;

        if (!name) return ['Missing name'];
        if (!pictureUrl) return ['Missing pictureUrl'];
        if (!pictureId) return ['Missing pictureId'];

        return [undefined, new UpdateUserDto(name, pictureUrl, pictureId)];
    }
}

File: ServerNET/Application/DTOs/Users/UpdateUserDto.cs
using System;

namespace Application.DTOs.Users;

public class UpdateUserDto(
    string name,
    string pictureId,
    string pictureUrl,
    string oldPublicId
)
{
    public string Name { get; set; } = name;
    public string PictureId { get; set; } = pictureId;
    public string PictureUrl { get; set; } = pictureUrl;
    public string OldPublicId { get; set; } = oldPublicId;
}
